InterfaceMATLAB/tsgGetDifferentiationWeights.m

```diff
@@ -0,0 +1,60 @@
+function [weights] = tsgGetDifferentiationWeights(lGrid, mX)
+%
+% [weights] = tsgGetDifferentiationWeights(lGrid, mX)
+%
+% it gives the weights for differentiation weights
+%
+% INPUT:
+%
+% lGrid: a grid list created by tsgMakeXXX(...) command
+%
+% mX: an array of size [num_x, dimensions]
+%     specifies the points where the derivative should be evaluated
+%     Note: do not confuse points here with the nodes of the grid
+%           here points are user specified points to evaluate the
+%           derivative
+%
+% OUTPUT:
+%
+% weights: an array of size [num_x, number_of_points]
```

Comment:
This is also incorrect, `weights` should hold `dimensions` somewhere in there.

Make it so that `f(points)' * weights` gives you the Jacobian in the case of `num_x == 1`. Then do the cell trick that I described above.